Goose in Docker with Cloudsql (#1)
travelaudience/docker-cloudsql-goose: README.md

# docker-cloudsql-goose

> Docker image for running [goose ](https://github.com/pressly/goose) migrations on a cloudsql database

Reviewer: ```suggestion
Docker image for running [goose ](https://github.com/pressly/goose) migrations in a cloudsql database
```

# docker-cloudsql-goose

> Docker image for running goose transactions on a cloudsql database

Reviewer: ```suggestion
Docker image for running [goose ](https://github.com/pressly/goose) migrations on a cloudsql database
```

> Docker image for running goose transactions on a cloudsql database

Reviewer: ```suggestion
Docker image for running goose migrations in a cloudsql database
```

# Usage

## Configuration (Local)

Parameters are passed as environment variables, so the image can easily be used from CircleCI (See below). For local development it is recommended to set up a `.env.local` file with the following contents:

```bash

> # Contents of the service account json file. (put everything on 1 line!)

Reviewer: why 1 line?

Reviewer: content of the env.local file instead of service account json file

# https://cloud.google.com/iam/docs/creating-managing-service-account-keys
CLOUDSQL_PROXY_SA={ XXXXXXXXX }
# The cloudsql instance you want to connect to
GCLOUD_SQL_INSTANCE=project:region:cloudsql_instance=tcp:localhost:port

# Goose & postgres connection configuration
POSTGRES_USER=postgres
POSTGRES_PASSWORD=XXXXXX
POSTGRES_PORT=5536
POSTGRES_DATABASE=goose_demo
GOOSE_DRIVER=postgres
```

## Build & Run (local)
```bash
# Build the image

> docker build -t cloudsql-goose . 

Reviewer: ```suggestion
docker build -t travelaudience/cloudsql-goose . 
```

# For local executions we mount the goose migration folder
docker run --rm -it --name goose \
    --env-file .env.local \

>     --mount type=bind,source="$(pwd)"/db/migrations,target=/db/migrations \

Reviewer: can you specify rw access and user as we did in docker-airflow-env docker-compose.yml file because goose will generate new files on host machine so we would like that uid:gid is the same as the host user.

Author: Done, but that does mean I can only use 777 directories (or mounted ones) to write temp files
